deployer/file_acquisition.py

```diff
@@ -104,29 +103,14 @@ def get_decrypted_file(original_filepath):
     if filename.startswith("enc-") or ("secret" in filename):
         # We must then determine if the file is using sops
         # sops files are JSON/YAML with a `sops` key. So we first check
-        # if the file is valid JSON/YAML, and then if it has a `sops` key
+        # if the file is valid JSON/YAML, and then if it has a `sops` key.
+        # Since valid JSON is also valid YAML by design, a YAML parser can read in JSON.
         with open(original_filepath) as f:
-
-            # FIXME: Right now we expect encrypted files to be JSON or YAML files, and
-            #        so we fail if these are not valid JSON/YAML. However in the
-            #        future, we may want to support encrypted files of other types
-            #        and we should update this section accordingly.
-            #
-            # Support the (clearly wrong) people who use .yml instead of .yaml
-            if ext == ".yaml" or ext == ".yml":
-                try:
-                    content = yaml.load(f)
-                except ScannerError:
-                    raise ScannerError(
-                        "We expect encrypted files to be valid JSON or YAML files."
-                    )
-            elif ext == ".json":
-                try:
-                    content = json.load(f)
-                except json.JSONDecodeError:
-                    raise json.JSONDecodeError(
-                        "We expect encrypted files to be valid JSON or YAML files."
-                    )
+            try:
+                content = yaml.load(f)
+            except ScannerError:
+                yield original_filepath
```

Comment:
Hmmm, it should not yield when the erroring - right? previously it raised an error on scanner error.

Reply:
In this case, we matched the file path name pattern - and should error rather than yield. Later we shouldn't.

```python
    else:
        # For a file that does not match our naming conventions for secrets, yield the
        # original path
        yield original_filepath
```

Reply:
I think I got confused when undoing my work in this commit https://github.com/2i2c-org/infrastructure/commit/7c676214729c81a5f5e0a7c59c15cc4799b52b17